api/src/main/java/org/apache/iceberg/util/BinaryUtil.java

```diff
@@ -28,12 +28,23 @@
   private BinaryUtil() {
   }
 
+  private static final ByteBuffer EMPTY_BYTE_BUFFER = ByteBuffer.allocate(0);
+
   /**
-   * Truncates the input byte buffer to the given length
+   * Truncates the input byte buffer to the given length.
+   * <p>
+   * We allow for a length of zero so that rows with empty string can be evaluated.
+   * Partition specs still cannot be created with a length of zero due to a constraint
+   * when parsing column truncation specs in org.apache.iceberg.MetricsModes.
```

Comment:
I tried using an `{@link}` here but the java docs couldn't find that class because it's in a different module.

Reply:
Pushing a version using @code. Thanks for the tip.